### PR TITLE
rest: add unit tests for read and list in rest pkg

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: server/rest/workspace/list_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	}, nil
+}
+
+func buildListRequest(workspace *restworkspacesv1alpha1.Workspace) *http.Request {
+	byteSlice, err := marshal.DefaultMarshal.Marshal(workspace)
+	Expect(err).NotTo(HaveOccurred())
+
+	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces", workspace.GetNamespace())
+
+	request, err := http.NewRequest("POST", url, bytes.NewReader(byteSlice))
```
Comment (sadlerap) [inline server/rest/workspace/list_test.go:99]: List requests are `GET`, not `POST`, right?  If so, replace this with `http.MethodGet`.

File: server/rest/workspace/read_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	}, nil
+}
+
+func buildGetRequest(workspace *restworkspacesv1alpha1.Workspace) *http.Request {
+	byteSlice, err := marshal.DefaultMarshal.Marshal(workspace)
+	Expect(err).NotTo(HaveOccurred())
+
+	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces", workspace.GetNamespace())
+
+	request, err := http.NewRequest("POST", url, bytes.NewReader(byteSlice))
```
Comment (sadlerap) [inline server/rest/workspace/read_test.go:99]: Same here.

**Review by sadlerap — CHANGES_REQUESTED** [review]

File: server/rest/workspace/list_test.go (hunk 1 above)
Comment (filariow) [reply server/rest/workspace/list_test.go:99]: Great catch! May you add the suggestion?

Comment (sadlerap) [reply server/rest/workspace/list_test.go:99]:
```suggestion
	request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(byteSlice))
```

File: server/rest/workspace/read_test.go (hunk 2 above)
Comment (sadlerap) [reply server/rest/workspace/read_test.go:99]:
```suggestion
	request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(byteSlice))
```

File: server/rest/workspace/read_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	return &coreworkspace.ReadWorkspaceResponse{
+		Workspace: nil,
+	}, nil
+}
+
+func buildGetRequest(workspace *restworkspacesv1alpha1.Workspace) *http.Request {
+	byteSlice, err := marshal.DefaultMarshal.Marshal(workspace)
+	Expect(err).NotTo(HaveOccurred())
+
+	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces", workspace.GetNamespace())
```
Comment (sadlerap) [inline server/rest/workspace/read_test.go:97]: This isn't the url for get requests, they need to have the name of the workspace:
```suggestion
	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces/%s", workspace.GetNamespace(), workspace.GetName())
```

**Review by sadlerap — CHANGES_REQUESTED** [review]

**Review by sadlerap — APPROVED** [review]

**Pull request merged**